docs/data-sources/template.md

```diff
@@ -10,7 +10,19 @@ description: |-
 
 An existing template on the Coder deployment.
 
-
+## Example Usage
+
+```terraform
+// Get a template on the provider's default organization by `id`
+data "coderd_template" "example-template" {
```

Comment:
Can't really think of a use-case for the template data source.. Open to suggestions to add here.

Reply:
May be it can be used similar to
https://github.com/coder/terraform-provider-coder/issues/220#issue-2281757765

Reply:
@ethanndickson, you may use the examples from the attached issue.

Reply:
I'm not sure I understand the use-case - where does the data from he template end up being used? In a workspace template?

Reply:
The `coder_workspace` data source from the `coder` provider already exposes `template_id`, `template_name` and `template_version` - if that's what you mean?

Reply:
Yes. I wondered if that same data is also exposed as part of the `coder_template` data source. If so, we can give an example of how to use it.

Reply:
What should that example be?

Reply:
The obvious example is if we had a `coder_workspace` resource, but until then it doesn't seem particularly useful.

Reply:
Something like
`data.coderd_template.example-template.id`
`data.coderd_template.example-template.template_name`
`data.coderd_template.example-template.active_version`
`data.coderd_template.example-template.latest_version`
`data.coderd_template.example-template.versions`

Reply:
I'm looking for an example use-case, what can a user actually do with those attributes once they've created the data source? What can they actually do with `example-template.id` once they have it?

Reply:
Ok. One use case is probably to copy settings from one template to another.

```diff
data "coderd_template" "ubuntu-old" {
 name = `ubuntu-old`
}

resource "coderd_template" "ubuntu-main" {
  name           = "ubuntu-main"
  description    = "The main template for developing on Ubuntu."
+ failure_ttl_ms = data.coderd_template.ubuntu-old.failure_ttl_ms 
  versions = [
    {
      name        = "stable-${var.TFC_CONFIGURATION_VERSION_GIT_COMMIT_SHA}"
      description = "The stable version of the template."
      directory   = "./stable-template"
    },
    {
      name        = "staging-${var.TFC_CONFIGURATION_VERSION_GIT_COMMIT_SHA}"
      description = "The staging version of the template."
      directory   = "./staging-template"
    }
  ]
}

```

Reply:
That's perfect - I'll add that in, thanks!

Reply:
I am happy that I was able to convey my intent 😊